src/main/java/org/springframework/data/elasticsearch/repository/support/AbstractElasticsearchRepository.java

```diff
@@ -203,15 +195,19 @@ public long count() {
 
 		Assert.notNull(entities, "Cannot insert 'null' as a List.");
 
-		operations.save(entities, getIndexCoordinates());
-		indexOperations.refresh(getIndexCoordinates());
+		Iterator<S> iterator = entities.iterator();
+		if (iterator.hasNext()) {
+			S entity = iterator.next();
+			operations.save(entities, getIndexCoordinates());
```

Comment:
How about returning `List<T>` from `save(…)` to investigate the index coordinates? 
In some cases, we've seen folks using `repo.saveAll(() -> Stream.of(…).map(…))` which isn't a proper iterable but it's used that way.